content/docs/guides/utf8.md

```diff
@@ -0,0 +1,107 @@
+---
+title: UTF-8 in Prometheus
+---
+
+# Introduction
+
+Versions of Prometheus before 3.0 required that metric and label names adhere to
+a strict set of character requirements. With Prometheus 3.0, all UTF-8 strings
+are valid names, but due to technical limitations it was not possible to create
+a totally seamless transition to the new character set.
+
+There may also be circumstances where users want to enforce the legacy character
+set, perhaps for compatibility with an older system or one that does not yet
+support UTF-8.
+
+# Instrumentation
+
+Currently, metrics created by the client_golang library will reject UTF-8 names
+by default. It is necessary to change the default validation scheme to allow
+UTF-8. The requirement to set this value will be removed in a future version of
+the common library.
+
+```golang
+import "github.com/prometheus/common/model"
+
+func init() {
+	model.NameValidationScheme = model.UTF8Validation
```

Comment:
Side idea: Should we also allow this per registry? 🤔

Reply:
Perhaps! This could be done as part of the solution to https://github.com/prometheus/client_golang/issues/1638